src/v2/cookbook/vue-route-change-handling.md

```diff
@@ -0,0 +1,376 @@
+---
+title: Vue Route Change Handling
+type: cookbook
+order: 10
+---
+
+## Simple Example
+
+While using [vue-router](https://router.vuejs.org/en/) we often need to refresh the data when a route or query changes while the same component is being reused in the page, to do so we can use a `:key` to force Vue to always re-create the components:
+
+```js
+Vue.component('OrderIndex', {
+  created() {
+    this.loadData();
+  },
+  methods: {
+    loadData() { /* logic to load data */ },
+  },
+});
+```
+
+```html
+<router-view :key="$route.fullPath"></router-view>
+```
+
+This will assure that when going from `/orders` to `/orders?page=2` or from `/orders?page=2` to `/orders?page=1`, a fresh instance of `OrderIndex` component will be provided, which will trigger the `created` method and download fresh data each time. There's also a subtle alternative to it:
+
+```html
+<router-view :key="$route.path"></router-view>
+```
+
+The main difference is that `/orders` and `/orders?page=2` would not re-create the components and data would not be reloaded, but going from `/orders` to `/invoices` with the same `OrderIndex` component would.
```

Comment:
I think this can be rephrased by saying _changes on the query eg: going from `/orders` to `/orders?page=2`, won't make up a different `key` and therefore, no data is fetched. However, going to a different route will keep the same behaviour.